x-pack/plugins/index_management/public/application/sections/home/template_list/template_details/tabs/tab_summary.tsx

```diff
@@ -53,9 +53,16 @@ export const TabSummary: React.FunctionComponent<Props> = ({ templateDetails })
 
   const numIndexPatterns = indexPatterns.length;
 
-  const {
-    core: { getUrlForApp },
-  } = useAppContext();
+  const [ilmPolicyLink, setIlmPolicyLink] = useState<string | undefined>();
```

Comment:
We should move this piece of state into `useUrlGenerator` and have it return  `{ url?: string }`. Then we can also remove `setLinkCallback`.

Reply:
I didn't think about using `setState` inside `useEffect`, great idea @jloleysens !